app/assets/javascripts/topology_preview.js

```diff
@@ -1,33 +1,34 @@
 var TopologyPreviewer = (function(){
-  var topologyGraph, svgGroup, data, root, tree, svgHeight, svgWidth;
+  var topologyGraph, presetData, style, svgGroup, data, root, tree, svgHeight, svgWidth,
+      currentStylesheet;
 
   var nodeIds         = 0,
       maxLabelLength  = 20,
-      duration        = 250,
-      endPointCounter = 0,
-      stepSize        = 20,
       ease            = 'cubic-out';
 
   var diagonal = d3.svg.diagonal()
-      .projection(function(d) { return [d.y, d.x]; });
+      .projection(function(d) { return [d.x, d.y]; });
 
   TopologyPreviewer.prototype = {
     preview: function(){
       clear();
 
-      svgHeight = viewerHeight();
-      svgWidth  = viewerWidth();
-      svg       = buildBaseSvg();
-      data      = [graphData()];
-      root      = data[0];
+      currentStylesheet = TopologyStylesheet[style];
```

Comment:
It reminds me of [this style](https://gist.github.com/antw/3c86b4df34281d768ca5) of creating JS classes. Perhaps that's what you were going for?